### DISCUSSION TITLE
Solved Games

### DISCUSSION THREAD

**Discussion by adamlawson22** [suragnair/alpha-zero-general#158] [Conceptual]
While training a model for solved games (TicTacToe, Connect4 and 3DTicTacToe), I noticed that when the model plays second against a human player, it plays almost randomly. Could this be that it assumes that the player who goes first is going to win, and therefore gives the same probability to each square, meaning it plays almost randomly?

**Comment by adamlawson22** [comment]
Is it possible to adapt the MCTS code to value a late lose more than an early lose?

**Comment by HaraldKorneliussen** [comment]
I don't see that it's necessary to do it there. You can just change the scoring in the getGameEnded method.

**Comment by adamlawson22** [comment]
The getGameEnded function returns 0 if the game isn't over, a 1 if current player won, a 0 if they lost and a small amount for a draw. How could this be adapted to give a higher score for a later loss?

**Comment by gms2009** [comment]
Question: Do you mean if you(maybe use some app on the web, I did see a perfect game for connect4 on the web)  play first,  your model has been trained so well it knows it is going to loss(after you played first move and every move), that means your model is (almost or really)perfect also?

**Comment by adamlawson22** [comment]
After enough training, the model will always win if it goes first. But if it goes second, it assumes it is playing an optimal player and knows it is going to lose, and returns equal probabilities for each move.

**Comment by suragnair** [comment]
Does this happen even after a few suboptimal moves? E.g. if the first player plays sub optimally for the first 10 moves?

**Comment by adamlawson22** [comment]
Yes, it doesn't even block moves that are obvious. If you allow it, in 3D tictactoe, to get two in a row and leave the square empty, it won't always play there, which would be a win.
